packages/client/src/monaco-converter.ts

```diff
@@ -423,19 +410,19 @@ export class MonacoToProtocolConverter {
         return markers.map(marker => this.asDiagnostic(marker));
     }
 
-    asCodeActionContext(context: monaco.languages.CodeActionContext): CodeActionContext {
+    asCodeActionContext(context: monaco.languages.CodeActionContext, diagnostics: Diagnostic[]): CodeActionContext {
         if (context === void 0 || context === null) {
             return context;
         }
-        const diagnostics = this.asDiagnostics(context.markers);
-        return CodeActionContext.create(diagnostics, Is.string(context.only) ? [context.only] : undefined);
+        // FIXME: CodeActionTriggerKind is missing
+        return CodeActionContext.create(diagnostics, Is.string(context.only) ? [context.only] : undefined, undefined);
```

Comment:
I currently don't see another way then saying `undefined` here, because there is no way to decide what to say here is right `Invoked: 1` or `Automatic: 2`? Is this even a FIXME, then?

Reply:
I did a PR on the vscode repo: https://github.com/microsoft/vscode/pull/149937

I don't see any reason for them to not merge it

Reply:
Superb! Ok, nevermind then.